sstables/sstables.cc

```diff
@@ -2200,26 +2200,32 @@ future<> sstable::snapshot(const sstring& dir) const {
     return _storage.snapshot(*this, dir);
 }
 
-future<> sstable::move_to_new_dir(sstring new_dir, generation_type new_generation, delayed_commit_changes* delay_commit) {
-    sstring old_dir = get_dir();
+future<> sstable::filesystem_storage::move(const sstable& sst, sstring new_dir, generation_type new_generation, delayed_commit_changes* delay_commit) {
```

Comment:
Note: after 725e5b119d2053a498924a324a471f79b426fd0f there is no real need for `new_generation`.
It might be currently used by unit tests, but we do without it.
- test_sstable_move can use same generation across multiple unique dir names
- same for test_sstable_move_replay
- test_sstable_move_exists_failure doesn't change the generation

Can be done as a follow, but also as a pre-cleanup to simplify the code prior to refactoring it